### ISSUE TITLE
Project dependencies should configure task dependencies

### ISSUE THREAD

**Issue by swankjesse** [square/wire#1768]
Today if I have this:

```
dependencies {
  protoPath project(":bookshelf-protos")
}
```

I also need this:

```
tasks.getByName("generateProtos").dependsOn(':bookshelf-protos:jar')
```

This sucks, I’d prefer the generateProtos dependency be added automatically.

**Comment by oldergod** [comment]
We added the dependency to the compile task but I wonder if that's be sufficient for 
```
dependencies {
  protoPath project(":bookshelf-protos")
}
```

**Comment by oldergod** [comment]
Released in 3.6 https://github.com/square/wire/blob/master/CHANGELOG.md#version-360
